src/CMakeLists.txt

```diff
@@ -24,6 +24,8 @@ mark_as_advanced(ICUBMAIN_COMPILE_SIMULATORS)
 option(ICUBMAIN_COMPILE_MODULES "Enable icub-main modules." ON)
 mark_as_advanced(ICUBMAIN_COMPILE_MODULES)
 
+option(UNITTEST_COMPILE "Enable unittest." OFF)
```

Comment:
It is a common convention in CMake projects to use the `BUILD_TESTING` option to control wheter the tests are built or not, see:
* https://github.com/robotology/how-to-export-cpp-library/blob/afb21efb655e7b1cc11636c3d42aad9e02fb626f/CMakeLists.txt#L100
* https://cliutils.gitlab.io/modern-cmake/chapters/testing.html

If you use that option name, then it would be obvious for CMake users which option to use to build the tests.

Reply:
Thanks, @traversaro  I will correct it ASAP

Reply:
@traversaro could you check also :

https://github.com/triccyx/icub-main/blob/8ccb44c1f9ac3ed60b1de5f4ea624a9575d94ca8/src/libraries/icubmod/embObjMultipleFTsensor/CMakeLists.txt#L30

I added this `if` because the plugin library prevents the unit testing. So I generate also the no plugin version of the library.
Not sure that this is "icub legacy"

Reply:
That seems fine, I would just remove `"$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${LIBRARY_TARGET_NAME}>"` from `target_include_directories`  that is not going to be used if the library is not installed. However, why the shared library can't be used for unit testing? Symbol visibility?

Reply:
> However, why the shared library can't be used for unit testing? Symbol visibility?

Yes, with shared I can not link

> That seems fine, I would just remove "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${LIBRARY_TARGET_NAME}>" from target_include_directories

Yes, you are right!